WIP-SQL/what-is-a-database-server.md

```diff
@@ -0,0 +1,61 @@
+# What is a database server?
+author: SebaRaba
+
+levels:
+
+  - beginner
+
+  - basic
+
+type: normal
+
+category: must-know
+
+links:
+
+  - '[More on database servers](http://ecomputernotes.com/fundamental/what-is-a-database/what-is-a-database-server)'
+
+---
+## Content
+
+A database server can be the *hardware* part of a database. A computer dedicated to perform database storage an retrieval. Such a hardware runs the database software  application (DBMS - database management system). It receives requests from the **client machines** and searches the collection for the **requested records**.
+
+As a software, it means the backend (also called instance) part of an database application. The backend handles tasks such as data analisys, storage, data manipulation and other non-user specific tasks. In general, database management systems provide database-server functionality. There are some DBMS[1] (such as MySQL) which rely exclusively on the *client-server model*[2].
+
+
+Note it's not mandatory that the actual data is hosted *on the same hardware server* as the database application.
+
+Considering a small or mid-sized setup. The hardware server will usually host the backend part of the software and the database. Think about a school's database system. The application will connect to the database via specific ports and use inter-process communication[3] to log into and access the data. The users will connect to the database via the client side of the application.
```

Comment:
Change to:
In a small application, something like a side project, the application and database will usually be hosted on the same computer, meaning they won't need to communicate over a network. 
A mid-sized application, such as an intranet or low-volume website, will most likely be hosted on a dedicated server while the database will be on a separate dedicated server. The application would then connect to the database via a Uniform Resource Locator.
In high-scale applications, the volume of data transactions can be so big that one single PC wouldn't be able to handle the work. In this case the software application may be running on hundreds of servers, and the database server will need to become a Cluster of servers, with one server doing the coordination, and other servers hosting portions of the database.